src/librustc_trans/trans/debuginfo/metadata.rs

```diff
@@ -796,12 +796,31 @@ pub fn type_metadata<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
             }
         }
         ty::TyBareFn(_, ref barefnty) => {
-            subroutine_type_metadata(cx, unique_type_id, &barefnty.sig, usage_site_span)
+            let fn_metadata = subroutine_type_metadata(cx,
```

Comment:
any reason that `fn_metadata` is defined up here but is only used after the `match` below?  (I note this largely because of that first `match` arm that returns, so I would have expected this `let` to come below the `match`)